src/plugins/intel_gpu/src/plugin/ops/non_max_suppression.cpp

```diff
@@ -189,6 +192,90 @@ static void CreateNonMaxSuppressionIEInternalOp(ProgramBuilder& p, const std::sh
     }
 }
 
-REGISTER_FACTORY_IMPL(internal, NonMaxSuppressionIEInternal);
+static void CreateNonMaxSuppressionOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v9::NonMaxSuppression>& op) {
+    cldnn::non_max_suppression::Rotation rotation = cldnn::non_max_suppression::Rotation::NONE;
+    validate_inputs_count(op, {2, 3, 4, 5, 6});
 
+    auto inputs = p.GetInputInfo(op);
+    std::vector<cldnn::input_info> reordered_inputs;
+    reordered_inputs.resize(inputs.size());
+
+    for (size_t portIndex = 0; portIndex < inputs.size(); portIndex++) {
+        auto inputDataType = cldnn::element_type_to_data_type(op->get_input_element_type(portIndex));
+        if ((portIndex == 2) && (inputDataType == cldnn::data_types::i64)) {
+            // GPU primitive supports only i32 data type for 'max_output_boxes_per_class' input
+            // so we need additional reorder if it's provided as i64
+            auto reorderPrimName = inputs[portIndex].pid + "_" + op->get_friendly_name() + ProgramBuilder::m_preProcessTag;
+            auto targetFormat = cldnn::format::get_default_format(op->get_input_partial_shape(portIndex).size());
+            auto preprocessPrim = cldnn::reorder(reorderPrimName,
+                                                 inputs[portIndex],
+                                                 targetFormat,
+                                                 cldnn::data_types::i32);
+            p.add_primitive(*op, preprocessPrim);
+            reordered_inputs[portIndex] = cldnn::input_info(reorderPrimName);
+        } else {
+            reordered_inputs[portIndex] = inputs[portIndex];
+        }
+    }
```

Comment:
There is some code duplication with `CreateNonMaxSuppressionIEInternalOp`. Could you refactor the code to reduce duplication?

Reply:
@isanghao Done refactoring. Thanks!